feat: Handle Substrate Block

kevinjanada opened the pull request:
## JIRA LINK
 https://blocksphere2020.atlassian.net/browse/DBIO-513?atlOrigin=eyJpIjoiNjkwMTJiMjMyMWM0NDVjYmIzNzc3MDYwZWI3YjIwYzMiLCJwIjoiaiJ9

 ## Changelog / Description
 - Create get and set block commands and handlers to remember the last
   block processed
 - Create syncblock method to handle missed blocks in case indexer goes
   offline
 - Refactor event handling in listenToEvents() method to handleEvent() method

agustinustheo commented:
Other than the switch case implementation, I have no problems as of yet, because I'm not well versed in Typescript yet. Probably going to give this approval with future notes of a probable refactor once I have a better grasp of the language.

agustinustheo reviewed (approved):
Dislike the switch case, will find alternatives tonight if possible.

kevinjanada committed:
feat: Handle Substrate Block

 ## JIRA LINK
 https://blocksphere2020.atlassian.net/browse/DBIO-513?atlOrigin=eyJpIjoiNjkwMTJiMjMyMWM0NDVjYmIzNzc3MDYwZWI3YjIwYzMiLCJwIjoiaiJ9

 ## Changelog / Description
 - Create get and set block commands and handlers to remember the last
   block processed
 - Create syncblock method to handle missed blocks in case indexer goes
   offline
 - Refactor event handling in listenToEvents() method to handleEvent() method

kevinjanada committed:
feat: Handle Service Created event improvement

- index the country and city of service
- use for loop in handle event in sync block to await each event handler

agustinustheo commented on src/substrate/substrate.handler.ts:
```diff
@@ -45,19 +47,73 @@ export class SubstrateService implements OnModuleInit {
+    const chunkSize = 200;
+    let iStart = lastBlockNumber;
+    let iEnd = currentBlockNumber;
+    if (iEnd - iStart > chunkSize) {
```
Maybe a comment here would make future devs understand easier

agustinustheo commented on src/substrate/substrate.handler.ts:
```diff
@@ -45,19 +47,73 @@ export class SubstrateService implements OnModuleInit {
+      await this.commandBus.execute(new SetLastBlockCommand(iEnd))
+
+      iStart = iEnd + 1;
+      iEnd = iEnd + chunkSize > endBlock ? endBlock : iEnd + chunkSize;
```
Please add another comment here for future devs.

agustinustheo commented on src/substrate/substrate.handler.ts:
```diff
@@ -87,28 +94,32 @@ export class SubstrateService implements OnModuleInit {
-              this.handleEvent(event);
-            });
-        });
+        const allEventRecords = await this.api.query.system.events.at(
```
Comments are missing

kevinjanada committed:
doc: Add comments to sync block algorithm
